causalml/feature_selection/filters.py

```diff
@@ -0,0 +1,370 @@
+"""
+Filter feature selection methods for uplift modeling
+
+- Currently only for classification problem: the outcome variable of uplift model is binary.
+"""
+
+import numpy as np
+import pandas as pd
+import statsmodels.api as sm
+from scipy import stats
+    
+class FilterSelect:
+    """A class for feature importance methods.
+    """
+
+    def __init__(self):
+        return
+
+    def _convert_treatment(self, data, T_name_before, treatment_key):
+        """
+        Convert text treatment labels into 0 (control) and 1 (treated).
+        """
+        T_dummies = pd.get_dummies(data[T_name_before])
+        T_dummies_treatment = T_dummies[treatment_key]   
+        return T_dummies_treatment
+
+
+    def _filter_F_one_feature(self, data, T_name, X_name, y_name): 
+        """
+        Conduct F-test of the interaction between treatment and one feature.
+        """
+        Y = data[y_name]
+        X = data[[T_name, X_name]]
+        X = sm.add_constant(X)
+        X['{}-{}'.format(T_name, X_name)] = X[[T_name, X_name]].product(axis=1)
+
+        model = sm.OLS(Y, X)
+        result = model.fit()
+
+        F_test = result.f_test(np.array([0, 0, 0, 1]))
+        F_test_result = pd.DataFrame({
+            'feature': X_name, # for the interaction, not the main effect
+            'method': 'F-statistic',
+            'score': F_test.fvalue[0][0], 
+            'p_value': F_test.pvalue, 
+            'misc': 'df_num: {}, df_denom: {}'.format(F_test.df_num, F_test.df_denom), 
+        }, index=[0]).reset_index(drop=True)
+
+        return F_test_result
+
+
+    def filter_F(self, data, T_name, X_name_list, y_name):
+        """
+        Rank features based on the F-statistics of the interaction.
```

Comment:
Could we add the args here?

Reply:
Updated "T_name" to a more interpretable name "treatment_indicator". This arg will only be used for internal functions. Args are added below.